builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -376,6 +375,82 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 	}
 }
 
+func resourceVSphereVirtualMachineUpdate(d *schema.ResourceData, meta interface{}) error {
+	hasChanges := false
+
+	// make config spec
+	configSpec := types.VirtualMachineConfigSpec{}
+
+	if d.HasChange("vcpu") {
+		configSpec.NumCPUs = d.Get("vcpu").(int)
+		hasChanges = true
+	}
+
+	if d.HasChange("memory") {
+		configSpec.MemoryMB = int64(d.Get("memory").(int))
+		hasChanges = true
+	}
+
+	// do nothing if there are no changes
+	if !hasChanges {
+		return nil
+	}
+
+	client := meta.(*govmomi.Client)
+	dc, err := getDatacenter(client, d.Get("datacenter").(string))
+	if err != nil {
+		return err
+	}
+	finder := find.NewFinder(client.Client, true)
+	finder = finder.SetDatacenter(dc)
+
+	vm, err := finder.VirtualMachine(context.TODO(), vmPath(d.Get("folder").(string), d.Get("name").(string)))
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] virtual machine config spec: %v", configSpec)
+
+	log.Printf("[INFO] Shutting down virtual machine: %s", d.Id())
+
+	task, err := vm.PowerOff(context.TODO())
+	if err != nil {
+		return err
+	}
+
+	err = task.Wait(context.TODO())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Reconfiguring virtual machine: %s", d.Id())
+
+	task, err = vm.Reconfigure(context.TODO(), configSpec)
+	if err != nil {
+		log.Printf("[ERROR] %s", err)
+	}
+
+	err = task.Wait(context.TODO())
+	if err != nil {
+		log.Printf("[ERROR] %s", err)
+	}
+
+	vm.PowerOn(context.TODO())
+
+	toolsRunning, err := vm.IsToolsRunning(context.TODO())
+	for !toolsRunning {
+		time.Sleep(1000 * time.Millisecond)
+		toolsRunning, err = vm.IsToolsRunning(context.TODO())
+	}
```

Comment:
I think it's a left over and I forgot to review it in detail. It should be redundant with WaitForIP(), since the vsphere clients won't show IPs without running tools either.
I'll remove it.